pkg/broker/handler/handler.go

```diff
@@ -102,3 +100,22 @@ func (h *Handler) receive(ctx context.Context, msg *pubsub.Message) {
 	}
 	msg.Ack()
 }
+
+func isNonRetryable(err error) bool {
+	// The following errors can be returned by ToEvent and are not retryable.
+	// TODO Should binding.ToEvent consolidate them and return the generic ErrCannotConvertToEvent?
+	return errors.Is(err, binding.ErrCannotConvertToEvent) || errors.Is(err, binding.ErrNotStructured) || errors.Is(err, binding.ErrUnknownEncoding) || errors.Is(err, binding.ErrNotBinary)
+}
+
+// Log the full message in debug level and a truncated version as an error in case the message is too big (can be as big as 10MB),
+func logEventConversionError(ctx context.Context, pm *pubsub.Message, err error, msg string) {
+	maxLen := 2000
+	truncated := pm
+	if len(pm.Data) > maxLen {
+		copy := *pm
+		copy.Data = copy.Data[:maxLen]
+		truncated = &copy
+	}
+	logging.FromContext(ctx).Debug(msg, zap.Any("message", pm), zap.Error(err))
+	logging.FromContext(ctx).Error(msg, zap.Any("message", truncated), zap.Error(err))
```

Comment:
Optional, but maybe suggest in the field key that the message data is truncated by calling it `message-truncated`?

Reply:
Good idea, done